src/geopathfinder/naming_conventions/yeoda_naming.py

```diff
@@ -360,6 +349,10 @@ def yeoda_tree(root, target_level=None, register_file_pattern=None):
                                 target_level=target_level,
                                 register_file_pattern=register_file_pattern)
 
+
+    if grid_pattern is not None:
```

Comment:
Will all files of the datacube path be loaded and filtered later on? Would it be possible to filter without loading all files first?

Reply:
you have the option to build the file register with 
build_smarttree()
or 
get_subtree_matching() / get_subtree_unique_rebased()

the build of the file_register is anyway not super-duper clean and neat - something for the future

Reply:
Sounds like a valid solution, thank you!